src/main/java/de/szut/lf8_project/lf8/project/dto/ProjectCreateDto.java

```diff
@@ -0,0 +1,74 @@
+package de.szut.lf8_project.lf8.project.dto;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
+import de.szut.employees.EmployeeAPI;
+import de.szut.lf8_project.exceptionHandling.InvalidDataException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.web.client.HttpClientErrorException;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.text.MessageFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+
+@Getter
+@Setter
+public class ProjectCreateDto {
+
+    @Min(0)
+    @NotNull
+    private Long customerId;
+
+    @Min(0)
+    @NotNull
+    private Long responsibleEmployeeId;
+
+    @Min(0)
+    @NotNull
+    private Long responsibleCustomerEmployeeId;
+
+    @Size(min = 1)
+    private List<Long> employees;
+
+    @Size(min = 1)
+    private List<String> skillSet;
+
+    @Size(min = 1)
+    @NotNull
+    private String description;
+
+    private LocalDate startDate;
+    private LocalDate endDate;
+
+    public ProjectCreateDto(Long customerId, Long responsibleEmployeeId, Long responsibleCustomerEmployeeId, List<Long> employees, List<String> skillSet, String description, LocalDate startDate, LocalDate endDate) {
+        this.customerId = customerId;
+        this.responsibleEmployeeId = responsibleEmployeeId;
+        this.responsibleCustomerEmployeeId = responsibleCustomerEmployeeId;
+        this.employees = employees;
+        this.skillSet = skillSet;
+        this.description = description;
+        this.startDate = startDate;
+        this.endDate = endDate;
+
+        if((startDate != null && endDate != null) && startDate.isAfter(endDate))
+            throw new InvalidDataException("The dates are not correct");
+
+        if(employees != null && !employees.isEmpty())
```

Comment:
Das brauchen wir nicht, da wir direkt danach durch alle Elemente der ArrayListe iterieren. Sollte es keine Elemente besitzen, also sie Leer sein, wird der Code im for-loop einfach geskiped
```suggestion
        if(employees != null)
```

Reply:
Yes we don't need this but skipping the loop using a if statement is actually more optimized than going into the loop and then realizing it's empty and then skipping it

Reply:
Hast du Testresultate, die das belegen?

Reply:
Dein fucking Glückstag xD

```java
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        ArrayList<Integer> arrayList = new ArrayList<>();
        // Approach 1: Check if ArrayList is empty before entering the for loop
        long startTime1 = System.nanoTime();
        if (arrayList != null && !arrayList.isEmpty()) {
            for (int i : arrayList) {
                // Process the elements of the ArrayList
            }
        }
        long endTime1 = System.nanoTime();
        long duration1 = endTime1 - startTime1;
        System.out.println("Approach 1 Duration: " + duration1 + " nanoseconds");

        // Approach 2: Directly enter the for loop without checking for emptiness first
        long startTime2 = System.nanoTime();
        if (arrayList != null) {
            for (int i : arrayList) {
                // Process the elements of the ArrayList
            }
        }
        long endTime2 = System.nanoTime();
        long duration2 = endTime2 - startTime2;
        System.out.println("Approach 2 Duration: " + duration2 + " nanoseconds");

        // Compare the durations
        if (duration1 < duration2) {
            System.out.println("Approach 1 is faster.");
        } else if (duration1 > duration2) {
            System.out.println("Approach 2 is faster.");
        } else {
            System.out.println("Both approaches have similar performance.");
        }
    }
}
```

Result:
Approach 1 Duration: 2405 nanoseconds
Approach 2 Duration: 35717 nanoseconds
Approach 1 is faster.

Reply:
Not luck, just a general rarely known information I knew. ;)

Reply:
<img width="425" alt="image" src="https://github.com/polyfish-lf8/lf8/assets/38872139/af1e6db1-2951-415c-8e04-c03e111b92de">

Reply:
<img width="987" alt="image" src="https://github.com/polyfish-lf8/lf8/assets/38872139/429d2e01-80d1-4ade-9d89-904953685814">